biodatasets/pubhealth/pubhealth.py

```diff
@@ -0,0 +1,190 @@
+        BigBioConfig(
+            name="pubhealth_source",
+            version=SOURCE_VERSION,
+            description="PUBHEALTH source schema",
+            schema="source",
+            subset_id="pubhealth",
+        ),
+        BigBioConfig(
+            name="pubhealth_bigbio_text",
+            version=BIGBIO_VERSION,
+            description="PUBHEALTH BigBio schema",
+            schema="bigbio_text",
+            subset_id="pubhealth",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "pubhealth_source"
+
+    def _info(self):
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "claim_id": datasets.Value("string"),
+                    "claim": datasets.Value("string"),
+                    "date_published": datasets.Value("string"),
+                    "explanation": datasets.Value("string"),
+                    "fact_checkers": datasets.Value("string"),
+                    "main_text": datasets.Value("string"),
+                    "sources": datasets.Value("string"),
+                    "label": datasets.Value("string"),
+                    "subjects": datasets.Value("string"),
+                }
+            )
+
+        # Using in pairs schema
+        elif self.config.schema == "bigbio_text":
+            features = schemas.text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        urls = _URLs[_DATASETNAME]
+        data_dir = Path(dl_manager.download_and_extract(urls))
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": os.path.join(data_dir, "PUBHEALTH/train.tsv"), "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": os.path.join(data_dir, "PUBHEALTH/test.tsv"), "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"filepath": os.path.join(data_dir, "PUBHEALTH/dev.tsv"), "split": "validation"},
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """Yields examples as (key, example) tuples."""
+
+        with open(filepath, encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(
+                csv_file, quotechar='"', delimiter="\t", quoting=csv.QUOTE_ALL, skipinitialspace=True
+            )
+            next(csv_reader, None)  # remove column headers
+            for id_, row in enumerate(csv_reader):
+                if split == "train" or split == "validation":
+                    if len(row) != 9:
+                        continue
```

Comment:
would it be possible to put in a comment about why this is needed? 
I did a little exploration and I did see one example that looked malformed (in train file and starts with `What's false: However, these guidelines affected` ... appears to be that the claim_id is missing)  ... maybe even record with line numbers in each file are being skipped?

Reply:
Added a logger to log the malformed lines.